swinging-cat-rx/src/animation.js

```diff
@@ -0,0 +1,116 @@
+import {Observable} from 'rxjs'
+
+/* CONSTATNST */
+
+const TRANSFORM = 'transform'
+
+const percents = [100, 87.5, 75, 62.5, 50, 37.5, 25, 12.5, 0]
+
+/* UTILS */
+
+function createIterations(percents, values) {
+  return percents.map((percent, i) => ({
+    percent: percent,
+    value: values[i]
+  }))
+}
+
+function createAnimationLoader(duration, delay = 0) {
+  return Observable
+    .interval(100)
+    .delay(delay)
+    .startWith(0)
+    .scan(x => x > duration * 10 ? 0 : x + 1, 0)
+    .map(x => x * 10 / duration)
+}
+
+function getAnimationValue($animationLoader, iterations) {
+  return $animationLoader.map(percent => {
+    for (let i = 0; i < iterations.length; i++) {
+      if (percent >= iterations[i].percent) {
+        return iterations[i].value
+      }
+    }
+    return iterations[iterations.length - 1].value
+  })
+}
+
+function getTransformTransition(time = 1, transitionType = 'linear') {
+  return { transition: `${TRANSFORM} ${time}s ${transitionType}`}
+}
+
+function createAnimation(duration, delay, transitionType, transionTime, iterations, modifyStyle) {
+  const $animationLoader = createAnimationLoader(duration, delay)
+  return {
+    [TRANSFORM]: getAnimationValue($animationLoader, iterations).map(modifyStyle),
+    ...getTransformTransition(transionTime, transitionType)
+  }
+}
+
+function createRotateAnimation(duration, delay, transitionType, values) {
+  const iterations = createIterations(percents, values)
+  return createAnimation(duration, delay, transitionType, 1, iterations, x => `rotate(${x}deg)`)
+}
+
+/* ANIMATIONS */
+
+export function faceAnimation(duration, delay, transitionType) {
```

Comment:
Question: why not to put each animation into the styles file where it is used, so that it is all together?